### PR TITLE
Create pull_request_template

### PR DESCRIPTION
**Pull request by nmnielsen**
### What Was The Problem? 
We want a standardized PR template to streamline context & knowledge sharing

### What Was The Solution?
Josh and I made a lightweight PR template and added it to this repo so that it is used as the default PR template in all other repositories in the organization.

### Where Does This Work Fall On The Good - Fast Spectrum?
Good & Fast - keeping the template lightweight to encourage adoption

### Any Additional Work Needed As A Result?
- Remove PR templates from other repos so that they all use this one
- At a future engineering round table, discuss potential changes/additions to the Template after it has been in use for a bit

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+### Are there any dependencies that need to be addresses before this PR is merged?
```
Comment (jedschneider) [inline .github/pull_request_template.md:1]: what kind of dependencies are we targeting here? software/library ones, people ones? Maybe it's intentionally vague to address either and I'm ok with that, but it is not clear to me if it is meant to be more specific.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+### What ticket(s) or other PRs does this relate to?
+
+
+### What was the problem or feature?
+
+
+### What was the solution?
+
+
+### Where does this work fall on the Good - Fast spectrum?
```
Comment (jedschneider) [inline .github/pull_request_template.md:13]: Is this to address the completeness of the solution or the urgency to review and publish?

File: .github/pull_request_template.md (hunk 1 above)
Comment (nmnielsen) [reply .github/pull_request_template.md:1]: We've intentionally left these pretty open ended in order to see what kind of use patterns people find natural, with the understanding that it's quick and easy to update the template in the future if we find it's not fitting our needs. We had both software and people dependencies in mind.

File: .github/pull_request_template.md (hunk 2 above)
Comment (jayzes) [reply .github/pull_request_template.md:13]: Oh, that's a good idea. Would it be worth asking two questions there?

Comment (nmnielsen) [reply .github/pull_request_template.md:13]: Primarily the completeness of the solution - particularly now as we try to build quickly, this is a space to call out specific technical corners that were cut in the name of expediency or that the solution aims for completeness. This gives the author the opportunity to self-evaluate and/or justify the coding decisions they've made, and gives the reviewer more context into the scope and intent of the change.

Comment (nmnielsen) [reply .github/pull_request_template.md:13]: (As for the "urgency to review", I think that should be communicated through other channels (slack, review requests, etc), as the description of a PR probably isn't the best place to stick a desperate plea for reviews of said PR)

**Pull request merged**